rq/worker.py

```diff
@@ -698,6 +698,7 @@ def perform_job(self, job, queue):
         try:
             with self.death_penalty_class(job.timeout or self.queue_class.DEFAULT_TIMEOUT):
                 rv = job.perform()
+            job.refresh()
```

Comment:
Could you please move `job.refresh()` to just before `job.save()` inside `handle_job_success()`? This should only be called if job needs to be persisted in Redis.

What would be even better is if we create `job.refresh_meta()` that uses `redis.hget()` to fetch only the `meta` field from Redis.